sql/mysqld.cc

```diff
@@ -8068,45 +8068,7 @@ mysqld_get_one_option(const struct my_option *opt, const char *argument,
   case (int)OPT_REPLICATE_REWRITE_DB:
   {
     /* See also OPT_REWRITE_DB handling in client/mysqlbinlog.cc */
-    const char* key= argument, *ptr, *val;
-
-    // Skipp pre-space in key
-    while (*key && my_isspace(mysqld_charset, *key))
-      key++;
-
-    // Where val begins
-    if (!(ptr= strstr(key, "->")))
-    {
-      sql_print_error("Bad syntax in replicate-rewrite-db: missing '->'");
-      return 1;
-    }
-    val= ptr+2;
-
-    // Skip blanks at the end of key
-    while (ptr > key && my_isspace(mysqld_charset, ptr[-1]))
-      ptr--;
-    if (ptr == key)
-    {
-      sql_print_error("Bad syntax in replicate-rewrite-db - empty FROM db");
-      return 1;
-    }
-    key= strmake_root(&startup_root, key, (size_t) (ptr-key));
-
-    /* Skipp pre space in value */
-    while (*val && my_isspace(mysqld_charset, *val))
-      val++;
-
-    // Value ends with \0 or space
-    for (ptr= val; *ptr && !my_isspace(&my_charset_latin1, *ptr) ; ptr++)
-    {}
-    if (ptr == val)
-    {
-      sql_print_error("Bad syntax in replicate-rewrite-db - empty TO db");
-      return 1;
-    }
-    val= strmake_root(&startup_root, val, (size_t) (ptr-val));
-
-    cur_rpl_filter->add_db_rewrite(key, val);
+    cur_rpl_filter->add_rewrite_db(argument);
```

Comment:
Ah, the invalid specification of `replicate-rewrite-db=db1` should be done within the `.test` file `mysqlbinlog` invocations rather than the `.opt` file (because `mysqld` won't start otherwise, as you noticed :))

So for example
```
--let $MYSQLD_DATADIR=`select @@datadir`
--let BINLOG_FILENAME= query_get_value(SHOW BINARY LOGS, Log_name, 1)
--let BINLOG_FILE_PARAM= $MYSQLD_DATADIR/$BINLOG_FILENAME

--echo # Invalid specification of replicate-rewrite-db should error
--echo # MYSQL_BINLOG BINLOG_FILE_PARAM --replicate-rewrite-db=db1 2> MYSQLD_VARDIR/tmp/mysqllbinlog.err.out
--error 1
--exec $MYSQL_BINLOG $BINLOG_FILE_PARAM --replicate-rewrite-db=db1 2> $MYSQLD_VARDIR/tmp/mysqlbinlog.err.out

# A quick use of include/search_pattern_in_file.inc or include/assert_grep.inc of mysqlbinlog.err.out to ensure the error message was written
```

Reply:
Done in new patch review for `sys_vars.replicate_rewrite_db.test`, using `--exec  $MYSQLD_CMD`, and the result file is updated for `main.mysqlbinlog.test` to reflect the same error message.